src/git.ts

```diff
@@ -81,3 +105,81 @@ export async function getCommitIfRightOrigin(repo: Repository, remoteOrigin: str
     return (parsedLocalRemoteOrigin.name === argsParsedRemoteOrigin.name && parsedLocalRemoteOrigin.owner === argsParsedRemoteOrigin.owner) ?
         (await getLastCommitDescription(repo))?.oid : null;
 }
+
+export function checkGitRemote(remoteOrigin: string): boolean {
+    return VALID_URL_REGEX.test(remoteOrigin);
+}
+
+export const TMP_DIR_PREFIX = "temp_rookout_";
+
+export async function cloneRemoteOriginWithCommit(repoUrl: string, commit: string, isDuplicate: boolean) {
+    // Assuming the last part of the remote origin url is the name of the repo.
+     const repoName = _.last(repoUrl.split("/")).replace(".git", "");
+     // If we have two of the same git remote with a different commit we want to create a sub directory.
+     const gitRoot = isDuplicate ? path.join(GIT_ROOT, `${TMP_DIR_PREFIX}${uuidv4()}`) : GIT_ROOT;
+     // Create the sub directory if needed.
+     if (isDuplicate) fs.mkdirSync(gitRoot);
+
+     // Getting the full path of the repo, including the repo name.
+     const repoDir = path.join(gitRoot, repoName);
+
+     // If the folder already exists we don't need to clone, just checkout.
+     const doesRepoExist = fs.existsSync(repoDir);
+
+     return new Promise<string>((resolve, reject) => {
+         exec(`${doesRepoExist ? "" : `cd "${gitRoot}" && git clone ${repoUrl} && `}cd "${repoDir}" && git checkout ${commit}`
+             , (error: any) => {
+                 if (error) {
+                     reject(error);
+                     return;
+                 }
+                 // Return the path of a successful clone
+                 resolve(repoDir);
+             });
+     });
+}
+// 10GB
+const MAX_GIT_FOLDER_SIZE = 10737418240;
+
+export async function isGitFolderBiggerThanMaxSize(): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+        getSize(GIT_ROOT, (err: Error, size: bigint) => {
+            if (err) {
+                reject(err);
+                return;
+            }
+
+            resolve(size > MAX_GIT_FOLDER_SIZE);
+        });
+    });
+}
+
+// Delete a folder even if it is not empty
+function deleteFolderRecursive(dirPath: string) {
```

Comment:
is there no package that does this? can we no just delete the folder and be done with it?